servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/AbstractEchoServerBasedHttpRequesterTest.java

```diff
@@ -52,23 +53,24 @@
 import static org.hamcrest.Matchers.hasToString;
 import static org.hamcrest.core.IsEqual.equalTo;
 
-public abstract class AbstractEchoServerBasedHttpRequesterTest {
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+abstract class AbstractEchoServerBasedHttpRequesterTest {
 
     @RegisterExtension
     static final ExecutionContextExtension CTX = ExecutionContextExtension.immediate().setClassLevel(true);
 
-    static ServerContext serverContext;
+    ServerContext serverContext;
```

Comment:
Static field change

Reply:
I'm not sure about the value of [MutableStaticState](https://pmd.github.io/pmd-6.35.0/pmd_rules_java_design.html#mutablestaticstate) check for tests. Making this field non-static forces us to remove `static` annotation from `@BeforeAll` method and add  `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` annotation for the class. Because this class is `abstract`, you don't know if any of its subclasses will ever have a per-test state.

To me, this validation creates more pain for tests in a long run. It doesn't for the scope of this PR, but we most likely will need to revisit and remove it later.